src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -65,49 +68,42 @@ public static String toQueryString(Map<String, String> params) {
      * @return A list of date strings in YYYY-MM-DD format corresponding to each day of the week to monitor.
      */
-    public static List<String> getDatesToCheckItineraryExistence(MonitoredTrip trip, boolean checkAllDays)
+    public static Set<ZonedDateTime> getDatesToCheckItineraryExistence(MonitoredTrip trip)
         throws URISyntaxException {
-        List<String> result = new ArrayList<>();
-        ZoneId zoneId = DateTimeUtils.getOtpZoneId();
+        Set<ZonedDateTime> datesToCheck = new HashSet<>();
         Map<String, String> params = trip.parseQueryParams();
 
         // Start from the query date, if available.
+        String startingDateString = params.get(DATE_PARAM);
         // If there is no query date, start from today.
-        String queryDateString = params.getOrDefault(
-            DATE_PARAM, DateTimeUtils.getStringFromDate(DateTimeUtils.nowAsLocalDate(), DEFAULT_DATE_FORMAT_PATTERN)
-        );
-        LocalDate queryDate = DateTimeUtils.getDateFromString(queryDateString, DEFAULT_DATE_FORMAT_PATTERN);
-
-        // Get the trip time.
-        ZonedDateTime queryZonedDateTime = ZonedDateTime.of(
-            queryDate, LocalTime.of(trip.tripTimeHour(), trip.tripTimeMinute()), zoneId
-        );
-
-        // Check the dates on days when a trip is active, or every day if checkAllDays is true,
-        // in a 7-day window starting from the query date.
-        for (int i = 0; i < 7; i++) {
-            ZonedDateTime probedDate = queryZonedDateTime.plusDays(i);
-            if (checkAllDays || trip.isActiveOnDate(probedDate)) {
-                result.add(DateTimeUtils.getStringFromDate(probedDate.toLocalDate(), DEFAULT_DATE_FORMAT_PATTERN));
+        LocalDate startingDate = startingDateString == null
+            ? DateTimeUtils.nowAsLocalDate()
+            : DateTimeUtils.getDateFromString(startingDateString, DEFAULT_DATE_FORMAT_PATTERN);
+        ZonedDateTime startingDateTime = trip.tripTime(startingDate);
+        // Get the dates to check starting from the query date and continuing through the full date range window.
+        for (int i = 0; i < ITINERARY_CHECK_WINDOW; i++) {
+            ZonedDateTime dateToCheck = startingDateTime.plusDays(i);
+            if (trip.isActiveOnDate(dateToCheck)) {
+                datesToCheck.add(dateToCheck);
             }
         }
 
-        return result;
+        return datesToCheck;
     }
 
     /**
      * Gets OTP queries to check non-realtime itinerary existence for the given trip.
      */
-    public static Map<String, String> getItineraryExistenceQueries(MonitoredTrip trip, boolean checkAllDays)
+    public static Map<ZonedDateTime, String> getItineraryExistenceQueries(MonitoredTrip trip)
         throws URISyntaxException {
         return getQueriesFromDates(
             excludeRealtime(trip.parseQueryParams()),
-            getDatesToCheckItineraryExistence(trip, checkAllDays)
+            getDatesToCheckItineraryExistence(trip)
```

Comment:
`checkAllDays` would be nice to not send extra OTP requests for days we don't need to check.